src/libraries/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs

```diff
@@ -330,169 +330,157 @@ internal unsafe struct SecPkgCred_ClientCertPolicy
+            void* logonID,
+            SafeSspiAuthDataHandle authdata,
+            void* keyCallback,
+            void* keyArgument,
+            ref CredHandle handlePtr,
+            out long timeStamp);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, CharSet = CharSet.Unicode, SetLastError = true)]
+        internal static unsafe partial int AcquireCredentialsHandleW(
+            string? principal,
+            string moduleName,
+            int usage,
+            void* logonID,
+            SCHANNEL_CRED* authData,
+            void* keyCallback,
+            void* keyArgument,
+            ref CredHandle handlePtr,
+            out long timeStamp);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, CharSet = CharSet.Unicode, SetLastError = true)]
+        internal static unsafe partial int AcquireCredentialsHandleW(
+            string? principal,
+            string moduleName,
+            int usage,
+            void* logonID,
+            SCH_CREDENTIALS* authData,
+            void* keyCallback,
+            void* keyArgument,
+            ref CredHandle handlePtr,
+            out long timeStamp);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, SetLastError = true)]
+        internal static unsafe partial int InitializeSecurityContextW(
+            ref CredHandle credentialHandle,
+            void* inContextPtr,
+            byte* targetName,
+            ContextFlags inFlags,
+            int reservedI,
+            Endianness endianness,
+            SecBufferDesc* inputBuffer,
+            int reservedII,
+            ref CredHandle outContextPtr,
+            ref SecBufferDesc outputBuffer,
+            ref ContextFlags attributes,
+            out long timeStamp);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, SetLastError = true)]
+        internal static unsafe partial int CompleteAuthToken(
+            void* inContextPtr,
+            ref SecBufferDesc inputBuffers);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, SetLastError = true)]
+        internal static unsafe partial int ApplyControlToken(
+            void* inContextPtr,
+            ref SecBufferDesc inputBuffers);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, SetLastError = true)]
+        internal static partial SECURITY_STATUS SspiFreeAuthIdentity(
+            IntPtr authData);
+
+        [GeneratedDllImport(Interop.Libraries.SspiCli, ExactSpelling = true, CharSet = CharSet.Unicode, SetLastError = true)]
+        internal static partial SECURITY_STATUS SspiEncodeStringsAsAuthIdentity(
+            string userName,
+            string domainName,
+            string password,
+            out SafeSspiAuthDataHandle authData);
+
+        // TODO: Switch to use GeneratedDllImport once we annotate blittable types used in interop in CoreLib (like Guid)
```

Comment:
We would annotate using the new `BlittableTypeAttribute` that a type is blittable and is designed to be used in interop scenarios.

Reply:
Why require another attribute rather than just following suite with the `unmanaged` constraint?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-7.3/blittable#detailed-design

Reply:
There is no annotation today that says that it is ok to use a given type in interop signatures and that the field layout of the type is stable. It is just very loosely mentioned in documentation. 

The idea behind `BlittableTypeAttribute` and friends was to introduce annotation that says this type has stable field layout and it is ok to use it in interop signatures.

However, I do like to idea of fixing the disconnect between `unmanaged` constraint and the blittable types now that you have mentioned it. Maybe it would be better to allow any type that passed `unmanaged` constraint in interop signatures and marshal it as blittable type (unless it has explicitly specified marshaler)?

Reply:
Unmanaged != blittable. Char and bool are both unmanaged but not always (or ever) blittable. This is covered in the StructMarshalling.md design doc.

Reply:
Right, `char` and `bool` and not blittable in the legacy interop.

The question to ask is whether we like it, and whether it would be better to drop the concept of blittable types as it exists in the source generated interop, and just stick with `unmanaged` types as defined by C#.

Reply:
> Maybe it would be better to allow any type that passed unmanaged constraint in interop signatures and marshal it as blittable type (unless it has explicitly specified marshaler)?

That's where my head was at, yeah.

Reply:
We can't really make the blittability support work seamlessly due to the built-in system still trying to marshal `bool` and `char` with their non-blittable representations. We'd have to have either some mechanism to opt-out a struct from the default built-in rules, or we'd have to generate a second struct that has the same fields, but with `byte` replacing `bool` (which can't be done reliably in any assembly other than the assembly the type lives in due to how ref assemblies might not have all fields available) and using `Unsafe.As` to convert between the two representations.

Reply:
Also, making the system an opt-out instead of an opt-in puts us back in the place where we may introduce a type in managed code that would have a useful representation in native code, but due to default marshalling would end up marshalling to native in a non-intuitive fashion. By making marshalling opt-in (even blittable cases), we can make sure that a type's default representation used in interop is in a fashion that the type owner/designer planned for.

Reply:
> We'd have to have either some mechanism to opt-out a struct from the default built-in rules

Yes, I think it may be a good idea to have "trust me nothing here requires marshalling" annotation for DllImports and function pointers.

Reply:
> Also, making the system an opt-out instead of an opt-in puts us back in the place where we may introduce a type in managed code that would have a useful representation in native code

The same can be said about C# `unmanaged`. Somebody may be accidently end up depending on the fact that the type is unmanaged, even though the type designer has not planned on the type being unmanaged.

The overarching point to investigate here is whether it would be worth it to harmonize interop and C# unmanaged.